tests/unit/py_proxy/views_test.py

```diff
@@ -105,86 +114,176 @@ def test_pdf_renders_parameters_in_pdf_template(
+            (
+                "/https://thirdparty.url/foo.pdf?bar=baz",
+                "http://localhost/pdf/https://thirdparty.url/foo.pdf?bar=baz",
+            ),
+        ],
+    )
+    def test_redirect_location(self, pyramid_config, requested_path, expected_location):
 
+        request = self._create_request(
+            request_url=requested_path,
+            thirdparty_url="https://thirdparty.url/foo.pdf",
+            content_type="application/pdf",
+            pyramid_config=pyramid_config,
+        )
+
+        redirect = views.content_type(request)
+
+        assert redirect.location == expected_location
 
-class TestContentTypeRoute:
     @pytest.mark.parametrize(
-        "content_type,redirect_location",
+        "content_type,redirect_url",
         [
-            ("application/pdf", "/pdf/http://thirdparty.url"),
-            ("application/x-pdf", "/pdf/http://thirdparty.url"),
-            ("text/html", "http://via.hypothes.is/http://thirdparty.url"),
+            ("application/pdf", "http://localhost/pdf/https://thirdparty.url"),
+            ("application/x-pdf", "http://localhost/pdf/https://thirdparty.url"),
+            ("text/html", "http://via.hypothes.is/https://thirdparty.url"),
         ],
     )
-    @httpretty.activate
     def test_redirects_based_on_content_type_header(
-        self, pyramid_request, content_type, redirect_location
+        self, pyramid_config, content_type, redirect_url
     ):
-        httpretty.register_uri(
-            httpretty.GET,
-            "http://thirdparty.url",
-            body="{}",
-            adding_headers={"Content-Type": content_type},
+        request = self._create_request(
+            request_url="/https://thirdparty.url",
+            thirdparty_url="https://thirdparty.url",
+            content_type=content_type,
+            pyramid_config=pyramid_config,
         )
 
-        result = views.content_type(pyramid_request)
+        result = views.content_type(request)
+
+        assert result.location == redirect_url
 
-        assert result.location == redirect_location
+    @pytest.mark.parametrize(
+        "query_param", ["via.request_config_from_frame", "via.open_sidebar"]
+    )
+    @pytest.mark.xfail(raises=AssertionError)
+    def test_does_not_pass_via_query_params_to_thirdparty_server(
+        self, pyramid_config, query_param
+    ):
+        request = self._create_request(
+            request_url="/https://thirdparty.url?"
+            "via.request_config_from_frame=lms.hypothes.is&via.open_sidebar=1",
+            thirdparty_url="https://thirdparty.url",
+            content_type="application/pdf",
+            pyramid_config=pyramid_config,
+        )
 
-    @pytest.fixture
-    def pyramid_request(self, pyramid_request):
-        pyramid_request.matchdict = {"url": "http://thirdparty.url"}
+        views.content_type(request)
 
-        def route_url(path, pdf_url):
-            return f"/{path}/{pdf_url}"
+        # pylint: disable-msg=E1101
```

Comment:
```suggestion
        # pylint: disable=no-member
```

You just need `disable=` not `disable-msg=` and we should always use the human-readable message names not the numeric codes.